frontend/src/pages/WorkflowitemDetails/WorkflowHistoryTab.js

```diff
@@ -2,59 +2,46 @@ import React from "react";
 import { connect } from "react-redux";
 
 import ScrollingHistory from "../Common/History/ScrollingHistory";
-import { fetchFirstWorkflowitemHistory, fetchNextWorkflowitemHistory } from "../WorkflowitemDetails/actions";
+import { fetchNextWorkflowitemHistoryPage } from "../WorkflowitemDetails/actions";
 
 function WorkflowitemHistoryTab({
   projectId,
   subprojectId,
   workflowitemId,
-  offset,
-  limit,
   events,
   nEventsTotal,
-  hasMore,
   isLoading,
   getUserDisplayname,
-  fetchFirstWorkflowitemHistory,
-  fetchNextWorkflowitemHistory,
-  isFirstHistoryCall
+  fetchNextWorkflowitemHistoryPage,
+  currentHistoryPage,
+  lastHistoryPage
 }) {
   return (
     <ScrollingHistory
       events={events}
       nEventsTotal={nEventsTotal}
-      hasMore={hasMore}
+      hasMore={currentHistoryPage !== lastHistoryPage}
       isLoading={isLoading}
       getUserDisplayname={getUserDisplayname}
-      // fetchNext={() => fetchFirstWorkflowitemHistory(projectId, subprojectId, workflowitemId, offset, limit)}
-      fetchNext={() => {
-        if (isFirstHistoryCall) {
-          fetchFirstWorkflowitemHistory(projectId, subprojectId, workflowitemId, offset, limit);
-        } else {
-          fetchNextWorkflowitemHistory(projectId, subprojectId, workflowitemId, offset, limit);
-        }
-      }}
+      fetchNext={() => fetchNextWorkflowitemHistoryPage(projectId, subprojectId, workflowitemId)}
       initialLoad={true}
     />
   );
 }
 
 function mapStateToProps(state) {
   return {
-    offset: state.getIn(["workflowitemDetails", "offset"]),
-    limit: state.getIn(["workflowitemDetails", "limit"]),
     events: state.getIn(["workflowitemDetails", "events"]),
```

Comment:
We should not pass ImmutableJS Objects to dumb components

Reply:
was converted to pure JS objects